--- git status ---
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   deploy/robots/g1_29dof/main.cpp
	modified:   deploy/robots/g1_29dof/src/State_Mimic.cpp
	modified:   source/unitree_rl_lab/unitree_rl_lab/tasks/locomotion/robots/g1/23dof/velocity_env_cfg.py

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/deploy/robots/g1_29dof/main.cpp b/deploy/robots/g1_29dof/main.cpp
index 9717e3a..be40418 100644
--- a/deploy/robots/g1_29dof/main.cpp
+++ b/deploy/robots/g1_29dof/main.cpp
@@ -38,7 +38,7 @@ int main(int argc, char** argv)
 
     init_fsm_state();
 
-    FSMState::lowcmd->msg_.mode_machine() = 4; // 29dof
+    FSMState::lowcmd->msg_.mode_machine() = 5; // 29dof
     if(!FSMState::lowcmd->check_mode_machine(FSMState::lowstate)) {
         spdlog::critical("Unmatched robot type.");
         exit(-1);
diff --git a/deploy/robots/g1_29dof/src/State_Mimic.cpp b/deploy/robots/g1_29dof/src/State_Mimic.cpp
index 4da579e..0931575 100644
--- a/deploy/robots/g1_29dof/src/State_Mimic.cpp
+++ b/deploy/robots/g1_29dof/src/State_Mimic.cpp
@@ -199,35 +199,7 @@ void State_Mimic::enter()
 void State_Mimic::run()
 {
     auto action = env->action_manager->processed_actions();
-    // Initial joint mapping for 29-dof robot
-    // joint_ids_map: [0, 6, 12, 1, 7, 13, 2, 8, 14, 3, 9, 15, 22, 4, 10, 16, 23, 5, 11, 17, 24, 18, 25, 19, 26, 20, 27, 21, 28]
-    // Remove actions that correspond to joint indices {13, 14, 20, 21, 26, 27} for 23-dof robot
-    // joint_ids_map: [0, 6, 12, 1, 7, 2, 8, 3, 9, 15, 22, 4, 10, 16, 23, 5, 11, 17, 24, 18, 25, 19, 28]
-    // Reorder actions according to new 23-dof robot mapping
-    // joint_ids_map: [0, 6, 12, 1, 7, 15, 22, 2, 8, 16, 23, 3, 9, 17, 24, 4, 10, 18, 25, 5, 11, 19, 26]
-
-    // Remove actions for ignored joints
-    // action.erase(action.begin() + 5,  action.begin() + 7);  // remove 13, 14
-    // action.erase(action.begin() + 11, action.begin() + 13); // remove 20, 21
-    // action.erase(action.begin() + 17, action.begin() + 19); // remove 26, 27
-
-    // // Reorder indices for 23-dof robot
-    // std::vector<int> reorder_indices = {0, 1, 2, 3, 4, 10, 11, 5, 6, 7, 8, 9, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22};
-    // std::vector<float> reordered_action(action.size());
-    // for (size_t i = 0; i < reorder_indices.size(); ++i) {
-    //     reordered_action[i] = action[reorder_indices[i]];
-    // }
-
-
-    // Initial joint mapping for 29-dof robot
-    // joint_ids_map: [0, 6, 12, 1, 7, 13, 2, 8, 14, 3, 9, 15, 22, 4, 10, 16, 23, 5, 11, 17, 24, 18, 25, 19, 26, 20, 27, 21, 28]
-    // Ignore actions that correspond to joints {13, 14, 20, 21, 26, 27} for 23-dof robot
-    std::vector<int> ignore_indices = {5, 8, 24, 25, 26, 27}; // indices of ignored joints in 29-dof action
-    for (int i(0); i < env->robot->data.joint_ids_map.size(); i++) {
-        int joint_id = env->robot->data.joint_ids_map[i];
-        if (std::find(ignore_indices.begin(), ignore_indices.end(), joint_id) != ignore_indices.end()) {
-            continue; // skip ignored joints
-        }
-        lowcmd->msg_.motor_cmd()[joint_id].q() = action[i];
+    for(int i(0); i < env->robot->data.joint_ids_map.size(); i++) {
+        lowcmd->msg_.motor_cmd()[env->robot->data.joint_ids_map[i]].q() = action[i];
     }
 }
\ No newline at end of file
diff --git a/source/unitree_rl_lab/unitree_rl_lab/tasks/locomotion/robots/g1/23dof/velocity_env_cfg.py b/source/unitree_rl_lab/unitree_rl_lab/tasks/locomotion/robots/g1/23dof/velocity_env_cfg.py
index 0da4156..e2aa1e5 100644
--- a/source/unitree_rl_lab/unitree_rl_lab/tasks/locomotion/robots/g1/23dof/velocity_env_cfg.py
+++ b/source/unitree_rl_lab/unitree_rl_lab/tasks/locomotion/robots/g1/23dof/velocity_env_cfg.py
@@ -252,7 +252,7 @@ class RewardsCfg:
 
     # -- base
     base_linear_velocity = RewTerm(func=mdp.lin_vel_z_l2, weight=-2.0)
-    base_angular_velocity = RewTerm(func=mdp.ang_vel_xy_l2, weight=-0.5)
+    base_angular_velocity = RewTerm(func=mdp.ang_vel_xy_l2, weight=-0.05)
     joint_vel = RewTerm(func=mdp.joint_vel_l2, weight=-0.001)
     joint_acc = RewTerm(func=mdp.joint_acc_l2, weight=-0.1e-7)
     action_rate = RewTerm(func=mdp.action_rate_l2, weight=-0.05)
@@ -261,7 +261,7 @@ class RewardsCfg:
 
     joint_deviation_arms = RewTerm(
         func=mdp.joint_deviation_l1,
-        weight=-0.1,
+        weight=-1.0,
         params={
             "asset_cfg": SceneEntityCfg(
                 "robot",
@@ -292,7 +292,7 @@ class RewardsCfg:
     )
 
     # -- robot
-    flat_orientation_l2 = RewTerm(func=mdp.flat_orientation_l2, weight=-5.0)
+    flat_orientation_l2 = RewTerm(func=mdp.flat_orientation_l2, weight=-10.0)
     base_height = RewTerm(func=mdp.base_height_l2, weight=-10, params={"target_height": 0.78})
 
     # -- feet